### PR TITLE
Priority order tasks even when using pools

### PR DESCRIPTION
**Pull request by tanelk**
When picking tasks to queue, the `scheduler_job` groups candidate task instances by their pools and picks tasks for queueing for each "pool group". 

This way tasks with lower priority could be queued before tasks with higher priority. This is demostrated in new UT `test_find_executable_task_instances_order_priority_with_pools` - before this PR `dummy3` and `dummy1` are queued instead of `dummy3` and `dummy2`.

Removing the grouping step (`pool_to_task_instances`) has minimal additional effect:

This log is removed:
https://github.com/apache/airflow/blob/0a99be741108470608a81964007aaf0a83f66a9f/airflow/jobs/scheduler_job.py#L376-L382

This can be logged more than once:
https://github.com/apache/airflow/blob/0a99be741108470608a81964007aaf0a83f66a9f/airflow/jobs/scheduler_job.py#L389-L392

If either of these is problematic, then with some additional work we can keep those.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by tanelk** [comment]
@ephraimbuddy could you take a look at this (small) bugfix. The bug has small impact, but it would be nice to clean up.

**Comment by tanelk** [comment]
Pinging @ashb for a review of this slight bugfix in the scheduler

File: airflow/jobs/scheduler_job.py [hunk 1]
```diff
@@ -355,141 +355,120 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
...
-                    self.log.warning("Tasks using non-existent pool '%s' will not be scheduled", pool)
+                if pool_name not in pools:
+                    self.log.warning("Tasks using non-existent pool '%s' will not be scheduled", pool_name)
                     starved_pools.add(pool_name)
                     continue
 
-                pool_total = pools[pool]["total"]
-                open_slots = pools[pool]["open"]
+                pool_total = pools[pool_name]["total"]
+                open_slots = pools[pool_name]["open"]
```
Comment (ashb) [inline airflow/jobs/scheduler_job.py]:
```python
                pool_stats = pools.get(pool_name)
                if not pool_stats:
                    self.log.warning("Tasks using non-existent pool '%s' will not be scheduled", pool_name)
                    starved_pools.add(pool_name)
                    continue
                pool_total = pool_stats["total"]
                open_slots = pool_stats["open"]
```

File: airflow/jobs/scheduler_job.py [hunk 2]
```diff
@@ -355,141 +355,120 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
...
-                    dag_active_tasks_map[dag_id] += 1
-                    task_concurrency_map[(task_instance.dag_id, task_instance.task_id)] += 1
-
-                pools[pool]["open"] = open_slots
+                executable_tis.append(task_instance)
+                open_slots -= task_instance.pool_slots
+                dag_active_tasks_map[dag_id] += 1
+                task_concurrency_map[(task_instance.dag_id, task_instance.task_id)] += 1
+
+                pools[pool_name]["open"] = open_slots
```
Comment (ashb) [inline airflow/jobs/scheduler_job.py]:
```suggestion
                pool_stats["open"] = open_slots
```

**Review by ashb — APPROVED** [review]

**Pull request merged**